### ISSUE TITLE
asynchronous suspension error, lost connection to device error

### ISSUE THREAD

**Issue by atakan351** [flutter/flutter#124927]
When i build on windows gave me this error.

Lost connection to device.
Exited (sigterm)

I tried;

flutter clean
flutter build
flutter pub get
flutter build windows

```
 #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:722:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1257:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:283:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:64:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>
```

flutter build -v ;

```
[  +12 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\INSTALL\INSTALL.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\INSTALL\INSTALL.tlog\INSTALL.lastbuildstate".
[        ] Done Building Project "D:\ATAKAN\Projects\myself\build\windows\INSTALL.vcxproj" (default targets).
[   +5 ms] Build succeeded.
[        ]     0 Warning(s)
[        ]     0 Error(s)
[        ] Time Elapsed 00:00:19.15
[  +24 ms] Building Windows application... (completed in 21,8s)
[ +330 ms] Observatory URL on device: http://127.0.0.1:60989/ILtHU2wu7eE=/
[   +3 ms] Caching compiled dill
[  +38 ms] Connecting to service protocol: http://127.0.0.1:60989/ILtHU2wu7eE=/
[ +108 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:60989/ILtHU2wu7eE=/.
[  +77 ms] DDS is listening at http://127.0.0.1:60992/Lvx5EKtYYgw=/.
[  +57 ms] Successfully connected to service protocol: http://127.0.0.1:60989/ILtHU2wu7eE=/
[  +13 ms] flutter: saat 2023-04-15 20:30:57.984461
[   +1 ms] flutter: geçen zaman : false
[  +12 ms] DevFS: Creating new filesystem on the device (null)
[  +23 ms] DevFS: Created new filesystem on the device (file:///C:/Users/Dinibh/AppData/Local/Temp/myselfc7b23d8c/myself/)
[   +2 ms] Updating assets
[ +152 ms] Manifest contained wildcard assets. Inserting missing file into build graph to force rerun. for more information see #56466.
[   +4 ms] Syncing files to device Windows...
[   +2 ms] Compiling dart to kernel with 0 updated files
[        ] Processing bundle.
[   +2 ms] <- recompile package:myself/main.dart 9246e8cc-623d-45c8-bd10-464f949efd61
[        ] <- 9246e8cc-623d-45c8-bd10-464f949efd61
[   +2 ms] Bundle processing done.
[ +157 ms] Updating files.
[        ] DevFS: Sync finished
[   +4 ms] Syncing files to device Windows... (completed in 167ms)
[   +4 ms] Synced 0.0MB.
[   +9 ms] <- accept
[   +3 ms] Connected to _flutterView/0x22e860c2430.
[   +4 ms] Flutter run key commands.
[   +5 ms] r Hot reload. 
[   +1 ms] R Hot restart.
[        ] h List all available interactive commands.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ]  Running with sound null safety 
[   +1 ms] An Observatory debugger and profiler on Windows is available at: http://127.0.0.1:60992/Lvx5EKtYYgw=/
[  +86 ms] flutter: deviceId->4C4C4544-0048-3310-8044-C7C04F474A33  
[   +1 ms] flutter: expiryDate :2023-04-16 08:14:58.000
[  +37 ms] The Flutter DevTools debugger and profiler on Windows is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:60992/Lvx5EKtYYgw=/
[+6305 ms] Service protocol connection closed.
[        ] Lost connection to device.
[   +1 ms] DevFS: Deleting filesystem on the device (file:///C:/Users/Dinibh/AppData/Local/Temp/myselfc7b23d8c/myself/)
[ +261 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[   +3 ms] "flutter run" took 31.864ms.
[ +194 ms] ensureAnalyticsSent: 191ms
[   +1 ms] Running 1 shutdown hook
[  +13 ms] Shutdown hooks complete
[   +1 ms] exiting with code 0
```

**Comment by atakan351** [comment]
@darshankawar / Windows 11

[√] Flutter (Channel stable, 3.7.11, on Microsoft Windows [Version 10.0.22621.1555], locale tr-TR)
    • Flutter version 3.7.11 on channel stable at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f72efea43c (6 days ago), 2023-04-11 11:57:21 -0700
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\Dinibh\AppData\Local\Android\sdk
    • Platform android-33-ext5, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.5.4)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.5.33530.505
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.1555]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 112.0.5615.86
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 112.0.1722.46

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

**Comment by darshankawar** [comment]
Do you get this error consistently everytime you run `flutter build windows` ?
If you create new Flutter app and run the same command, do you still get same error ?
Do you have any anti-virus enabled ?

**Comment by arunspacek** [comment]
> Do you get this error consistently everytime you run `flutter build windows` ? If you create new Flutter app and run the same command, do you still get same error ? Do you have any anti-virus enabled ?

Hi Darshan. I'm having a similar problem. Everything is fine when I executed 'flutter doctor'.
I was able to build the windows app, but whenever I hot reload/hot restart, I get 
"Lost connection to device.
Exited(sigterm)"
in the Debug console.
It only happens when I use vscode. I had no problems while running the app using Android Studio.
The flutter version I'm using is 3.7.0. OS: Windows 11 Home. SDK: Windows 11. I have McAfee enabled.
I whitelisted code.exe in my antivirus software, but the issue still persists.
